rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -925,30 +971,33 @@ def _prepare_sequence_layers(self) -> None:
         self._tf_layers["sparse_dropout"] = tf_layers.SparseDropout(
             rate=self.config[DROPRATE]
         )
-        self._tf_layers["sparse_to_dense.text"] = self._create_sparse_dense_layer(
-            self.data_signature["text_features"],
-            "text",
-            self.config[C2],
-            self.config[DENSE_DIM]["text"],
-        )
-        self._tf_layers["sparse_to_dense.label"] = self._create_sparse_dense_layer(
-            self.data_signature["label_features"],
-            "label",
-            self.config[C2],
-            self.config[DENSE_DIM]["label"],
-        )
+        if "text_features" in self.data_signature:
+            self._tf_layers["sparse_to_dense.text"] = self._create_sparse_dense_layer(
+                self.data_signature["text_features"],
+                "text",
+                self.config[C2],
+                self.config[DENSE_DIM]["text"],
+            )
+        if "label_features" in self.data_signature:
```

Comment:
shouldn't this be: `if self.config[INTENT_CLASSIFICATION]:`? and if `label_features` are not there, raise?

Reply:
I'm creating response selector PR to this branch, will add error there